name.abuchen.portfolio/src/name/abuchen/portfolio/model/Bookmark.java

```diff
@@ -42,11 +48,63 @@ public boolean isSeparator()
 
     public String constructURL(Security security)
     {
-        String url = pattern.replace("{tickerSymbol}", encode(security.getTickerSymbol())); //$NON-NLS-1$
-        url = url.replace("{isin}", encode(security.getIsin())); //$NON-NLS-1$
-        url = url.replace("{wkn}", encode(security.getWkn())); //$NON-NLS-1$
-        url = url.replace("{name}", encode(security.getName())); //$NON-NLS-1$
+        boolean replacementDone = Boolean.FALSE;
+        String url = pattern;
+        Class<? extends Security> securityClass = security.getClass();
+        HashMap<String, String> types = new HashMap<String, String>();
+        TreeMap <Integer,Entry<String, String>> replaceTypes = new TreeMap<Integer,Entry<String, String>>();
+
+        
+        
+        types.put("{tickerSymbol}", "getTickerSymbol");
```

Comment:
Vielleicht ist es auch einfacher die Werte direkt in die Map zu setzen - statt den Umweg über die Methode. Es ist ja nicht so als wären das teuere Methodenaufrufe die man nur machen möchte wenn tatsächlich eine Ersetzung stattfindet. Die Werte sind ja da. 😄